### PR TITLE
chore: skip put object suite on legacy apiserver tests

### PR DESCRIPTION
**Pull request by nvinuesa**
This patch removes the putObjectSuite (which are testing.Factory tests) from the apiserver, while documenting them in the `TestStub()` method, following the steps of https://github.com/juju/juju/pull/18441.

_Note: this work is needed to land https://github.com/juju/juju/pull/18504 when it's ready._

### CODE REVIEW

File: apiserver/objects_legacy_test.go [hunk 1]
```diff
@@ -4,32 +4,20 @@
 package apiserver_test
```
Comment (SimonRichardson) [inline apiserver/objects_legacy_test.go:4]: Delete this file. I've removed it as well.

File: apiserver/facades/agent/provisioner/provisioninginfo_test.go [hunk 2]
```diff
@@ -628,67 +627,6 @@ func (s *withoutControllerSuite) TestProvisioningInfoWithUnsuitableSpacesConstra
 	c.Assert(result, jc.DeepEquals, expected)
 }
 
-func (s *withoutControllerSuite) TestProvisioningInfoWithLXDProfile(c *gc.C) {
```
Comment (SimonRichardson) [inline apiserver/facades/agent/provisioner/provisioninginfo_test.go:631]: Make a note of all the LXD profile tests, as we will want to reinstate them at some point.

**Review by SimonRichardson — APPROVED** [review]

**Review by jack-w-shaw — APPROVED** [review]

File: apiserver/facades/agent/provisioner/provisioninginfo_test.go (hunk 2 above)
Comment (nvinuesa) [reply apiserver/facades/agent/provisioner/provisioninginfo_test.go:631]: Sorry this one is actually OK, I won't remove it on this commit, it'll be part of another cleanup.

**Pull request merged**